### PR TITLE
Remove Nix

### PR DESCRIPTION
**Pull request by fzakaria**
Feedback from @rogerbinns and others was to go "vanilla" python project. 

Therefore remove Nix, Poetry and go back to published versions of dependencies such as apsw and lief

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,23 +66,18 @@ erDiagram
...
 
 ## Installation
-This repository can easily be installed, you simply need to have [Nix or NixOS](https://nixos.org) installed.
-
 ```console
-❯ nix run github:fzakaria/sqlelf /usr/bin/python3 -- \
+❯ python3 -m venv venv
+❯ source venv/bin/activate
+❯ pip install .
+❯ sqlelf /usr/bin/python3 -- \
```
Comment (markrwilliams) [inline README.md:73]: nit: it'd be nice to change this to either `/usr/bin/env python3` or `$VIRTUAL_ENV/bin/python3` because user's `python3` might be anywhere on the file system

File: README.md [hunk 2]
```diff
@@ -249,19 +244,15 @@ WHERE
...
+> make fmt
 ```
 
-A helping `Makefile` is provided to run all the _linters_ and _formatters_.
+You may want to install the package in _editable mode_ as well to make development easier
 
 ```console
-> make lint
-> make fmt
+> pip install --editable ".[dev]"
```
Comment (markrwilliams) [inline README.md]: this should go at the top of the Development section because it installs the dependencies used by `make lint` and `make fmt`

File: sqlelf/cli.py [hunk 3]
```diff
@@ -54,45 +50,16 @@ def start(args=sys.argv[1:], stdin=sys.stdin):
         ),
     )
     # Filter the list of filenames to those that are ELF files only
-    filenames = list(
-        filter(lambda f: os.path.isfile(f) and Binary.is_elf(f), filenames)
-    )
+    filenames = list(filter(lambda f: os.path.isfile(f) and lief.is_elf(f), filenames))
```
Comment (markrwilliams) [inline sqlelf/cli.py]:
```python
filenames = [f for f in filenames if os.path.isfile(f) and lief.is_elf(f)]
```

File: sqlelf/cli.py [hunk 4]
```diff
@@ -54,45 +50,16 @@ def start(args=sys.argv[1:], stdin=sys.stdin):
...
 
     # If none of the inputs are valid files, simply return
     if len(filenames) == 0:
         sys.exit("No valid ELF files were provided")
 
-    binaries: list[Binary] = [Binary(filename) for filename in filenames]
```
Comment (markrwilliams) [inline sqlelf/cli.py:65]: is this related to removing nix?

File: sqlelf/sql.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+import apsw
+import apsw.shell
+import lief
+
+from sqlelf import ldd
+from sqlelf.elf import dynamic, header, instruction, section, strings, symbol
+
+
+class SQLEngine(object):
```
Comment (markrwilliams) [inline sqlelf/sql.py]: don't inherit from `object` - that was important in python 2 but is redundant in python 3

File: sqlelf/sql.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    def __init__(self, binaries: list[lief.Binary], recursive=False) -> None:
```
Comment (markrwilliams) [inline sqlelf/sql.py]: nit: it's better for testing to move initialization logic to a module-level function and reduce the `__init__` to simple assignment (which makes `dataclasses` a good fit) -- it makes testing a lot easier!

in this case that'd look something like:

```python

@dataclass
class SQLEngine:
    connection: apsw.Connection

...

def make_sql_engine(binaries: list[lief.Binary], recursive=False) -> SQLEngine:
    connection = apsw.Connection(':memory:')
    ...
    return SQLEngine(connection)
```

with that change, anything you'd want to test can accept a `SQLEngine` backed by an in-memory database, without having to set up a directory tree of ELF objects.

File: sqlelf/elf/section.py [hunk 7]
```diff
@@ -5,23 +5,22 @@
...
     def generator() -> Iterator[dict[str, Any]]:
         for binary in binaries:
             # super important that these accessors are pulled out of the tight loop
             # as they can be costly
-            binary_path = binary.path
+            binary_name = binary.name
             for section in binary.sections:
                 yield {
-                    "path": binary_path,
+                    "path": binary_name,
```
Comment (markrwilliams) [inline sqlelf/elf/section.py:19]: nit: why `binary_name` instead of path? is it just the base name, not a full path? that's odd given dictionary's key remains `path`

**Review by markrwilliams — COMMENTED** [review]
please move the `pip install -e .[dev]` part of the readme above the makefile stuff per the comment.

everything else is optional!

File: README.md (hunk 1 above)
Comment (fzakaria) [reply README.md:73]: are you talking about the `sqlelf` invocation or the `-m venv` one?

File: sqlelf/cli.py (hunk 4 above)
Comment (fzakaria) [reply sqlelf/cli.py:65]: I downgraded LIEF as part of the nix removal and it changed the API a bit.

File: sqlelf/elf/section.py (hunk 7 above)
Comment (fzakaria) [reply sqlelf/elf/section.py:19]: it was called name on the object model even though its the path.
It's just an annoying thing about LIEF

File: README.md (hunk 1 above)
Comment (markrwilliams) [reply README.md:73]: sorry, the `sqlelf` invocation because the first command line argument is the absolute path of a python executable
